pkg/kubelet/dockershim/network/plugins.go

```diff
@@ -250,17 +255,36 @@ func getOnePodIP(execer utilexec.Interface, nsenterPath, netnsPath, interfaceNam
 }
 
 // GetPodIP gets the IP of the pod by inspecting the network info inside the pod's network namespace.
-func GetPodIP(execer utilexec.Interface, nsenterPath, netnsPath, interfaceName string) (net.IP, error) {
-	ip, err := getOnePodIP(execer, nsenterPath, netnsPath, interfaceName, "-4")
-	if err != nil {
-		// Fall back to IPv6 address if no IPv4 address is present
-		ip, err = getOnePodIP(execer, nsenterPath, netnsPath, interfaceName, "-6")
-	}
-	if err != nil {
-		return nil, err
-	}
+// TODO (khenidak). The "primary ip" in dual stack world does not really exist. For now
+// we are defaulting to v4 as primary
+func GetPodIPs(execer utilexec.Interface, nsenterPath, netnsPath, interfaceName string) ([]net.IP, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.IPv6DualStack) {
+		ip, err := getOnePodIP(execer, nsenterPath, netnsPath, interfaceName, "-4")
+		if err != nil {
+			// Fall back to IPv6 address if no IPv4 address is present
+			ip, err = getOnePodIP(execer, nsenterPath, netnsPath, interfaceName, "-6")
+		}
+		if err != nil {
+			return nil, err
+		}
 
-	return ip, nil
+		return []net.IP{ip}, nil
+	} else {
+		list := make([]net.IP, 0)
+		var err4, err6 error
+		if ipv4, err4 := getOnePodIP(execer, nsenterPath, netnsPath, interfaceName, "-4"); nil != err4 {
+			list = append(list, ipv4)
+		}
+
+		if ipv6, err6 := getOnePodIP(execer, nsenterPath, netnsPath, interfaceName, "-6"); nil != err6 {
+			list = append(list, ipv6)
+		}
+
+		if 0 == len(list) {
```

Comment:
it can be the case that pods only have on ip family despite dual-stack is enable, right?

Reply:
invert conditional

Reply:
@aojea no. `dualstack` validates that it is always dual cidr, dual IP etc..